### ISSUE TITLE
lx_emul usb: endpoint reset in FLUSH_TRANSFERS request may hang up device

### ISSUE THREAD

**Issue by skalk** [genodelabs/genode#5050]
People reported different scenarios in which an USB device passed-through to a virtual machine guest OS in VirtualBox hung up after certain usage scenarios. I could reproduce a hang-up of a FIDO2 authentication token repeatedly. In that scenario the device wasn't useable anymore after several FLUSH_TRANSFERS requests made by the Qemu XHCI model used in our VirtualBox port.
The XHCI model is the only client, which is using this kind of request, and if I understand the commit history of the USB session correctly, it is the reason why FLUSH_TRANFERS was ever introduced.
When looking at the context of the Qemu XHCI model code that calls this request, it seems that only ongoing transfers should be stopped. In the past we used within the Linux kernel driver port `usb_hcd_flush_endpoint` therefore, which cancels already enqueued transfers, but does even more. When introducing the USB user-level devio API, this call could not be used anymore without violating the bookkeeping of the devio layer. Instead all requests where canceled and an endpoint reset function got called on the devio API level, to keep close at the semantic of `usb_hcd_flush_endpoint`. But this reset function does even more, and stops the endpoint in between, and seems to render some devices at certain points useless. 
To sum it up, we do too much sensitive endpoint handling at a point where the client (the Qemu XHCI model) only wants to cancel ongoing transfers. Therefore, the reset endpoint function should get removed simply, but the cancelation of URBs submitted should be enough.
